features/mbedtls/platform/src/shared_rng.cpp

```diff
@@ -0,0 +1,75 @@
+/*
+ *  shared_rng.cpp
+ *
+ *  Copyright (C) 2019, Arm Limited, All Rights Reserved
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *  not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+#include "shared_rng.h"
+
+#if defined(MBEDTLS_SSL_CONF_RNG)
+
+#include "mbed_trace.h"
+
+mbedtls_hmac_drbg_context global_hmac_drbg;
+mbedtls_entropy_context global_entropy;
+static bool is_initialized = false;
+
+int init_global_rng()
+{
+    mbedtls_entropy_init(&global_entropy);
+    mbedtls_hmac_drbg_init(&global_hmac_drbg);
+
+    int ret = mbedtls_hmac_drbg_seed(&global_hmac_drbg, mbedtls_md_info_from_type(MBEDTLS_MD_SHA256),
+                                 mbedtls_entropy_func, &global_entropy, NULL, 0);
+
+    if (ret != 0) {
+        tr_error(" init_global_rng failed! mbedtls_hmac_drbg_seed returned -0x%x", -ret);
+        free_global_rng();
+    } else {
+        is_initialized = true;
+    }
+
+    return ret;
+This conversation was marked as resolved by jarvte
```

Comment:
In commit "Added new global rng, needed for MbedTLS optimisations", please remove this line, otherwise the commit won't build.

```diff
     return ret;
- This conversation was marked as resolved by jarvte
 }
```

Reply:
I will fix this. Didn't notice while merging

Reply:
Fixed